lib/AST/Decl.cpp

```diff
@@ -5891,6 +5891,39 @@ Expr *swift::findOriginalPropertyWrapperInitialValue(VarDecl *var,
   return initArg;
 }
 
+/// Writes a tuple expression where each element is either `nil` or another such
+/// tuple of nils.
+/// This comes up when printing default arguments for memberwise initializers
+/// that were created implicitly.
+/// For example, this var:
+/// ```
+/// var x: (Int?, (Int?, Int?, ()))
+/// ```
+/// will produce `(nil, (nil, nil, ()))`
+static void writeTupleOfNils(TupleType *type, ASTContext &ctx,
+                             llvm::raw_ostream &os) {
+  os << '(';
+  for (unsigned i = 0; i < type->getNumElements(); ++i) {
+    auto &elt = type->getElement(i);
+    if (elt.hasName()) {
+      os << elt.getName().str() << ": ";
+    }
+
+    if (elt.getType()->getAnyNominal() == ctx.getOptionalDecl()) {
+      os << "nil";
+    } else if (auto tupleTy = elt.getType()->getAs<TupleType>()) {
```

Comment:
Change `else if(... getAs())` to a `else { ... castTo()` and remove the llvm_unreachable below